x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/ServiceUtils.java

```diff
@@ -361,6 +419,25 @@ public static <E extends Enum<E>> E extractOptionalEnum(
         return null;
     }
 
+    public static Boolean extractRequiredBoolean(
+        Map<String, Object> map,
+        String settingName,
+        String scope,
+        ValidationException validationException
+    ) {
+        Boolean requiredField = ServiceUtils.removeAsType(map, settingName, Boolean.class);
+
+        if (requiredField == null) {
+            validationException.addValidationError(ServiceUtils.missingSettingErrorMsg(settingName, scope));
+        }
+
+        if (validationException.validationErrors().isEmpty() == false) {
```

Comment:
Same comment about `initialValidationErrorCount`

Reply:
Looks like that method (`extractRequiredBoolean`) is not used anywhere (I think I added it early on because I thought I needed it somewhere...) -- I can just remove this method to keep the code clean.